src/ZipkinBundle/Components/HttpClient/ContextRetrievers/InRequest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace ZipkinBundle\Components\HttpClient\ContextRetrievers;
+
+use Zipkin\Propagation\TraceContext;
+use ZipkinBundle\Components\HttpClient\ContextRetriever;
+
+/**
+ * InRequest stores and retrieves the tracing context from in the
+ * request options, making it explicit but embracing event loop
+ * models.
+ */
+final class InRequest implements ContextRetriever
+{
+    const TRACE_CONTEXT_EXTRA_KEY = 'zipkin_bundle_trace_context';
+
+    public static function wrapOptions(TraceContext $context, array $options = []): array
+    {
+        return $options + ['extra' => $options['extra'] + [self::TRACE_CONTEXT_EXTRA_KEY => $context]];
```

Comment:
not sure what extra means here. in brave we use extra for things like baggage which are in addition to the context. the context includes extra.

Reply:
In this case `extra` is the idiomatic symfony way to add user data into the "request". You can put a key and whatever as value. It is the only entrypoint I see to pass the context in the request call signature and consume it inside the tracing code for that request call.

Reply:
makes sense